pallets/creditcoin/src/tests.rs

```diff
@@ -3105,3 +3130,93 @@ fn register_address_v2_should_error_when_signature_is_invalid() {
 		);
```

Comment:
The corresponding `register_address_should_error_when_signature_is_invalid()` expects an `InvalidSignature` error here, which comes from this snippet:

```
                        // Extract public key of keypair used to sign the address of the caller
                        let signature = <[u8; 65]>::from(ownership_proof);
                        let raw_pubkey = secp256k1_ecdsa_recover_compressed(&signature, message)
                                .map_err(|_| Error::<T>::InvalidSignature)?;
```


I'm not sure what the equivalent of this is in register_address_v2.

Reply:
This is also impacted by the discussion above about `MalformedExternalAddress`. Now that we explicitly check the for the other error conditions we can properly return InvalidSignature as you mention here